collect_app/src/test/java/org/odk/collect/android/activities/MainActivityTest.java

```diff
@@ -197,4 +200,18 @@ public void manageFilesButtonTest() throws Exception {
         assertEquals(FileManagerTabs.class.getName(),
                 shadowIntent.getIntentClass().getName());
     }
+
+    /**
```

Comment:
I had never seen this test file! Well done on finding it. I think it's good that you've followed the style set by the file but it's a little unusual compared to the tests we'd usually write. Generally we try and be as descriptive as possible in our test names rather than using JavaDoc. For example I would have named your test `pressingConfigureQRCode_launchesScanQRCodeActivity`. I like the test itself.

It also looks like test doesn't set up the application for the `MainMenuActivity` correctly - this is why you were getting a `null` `adminSharedPreferences`. The activity's `onCreate` has this code:

```java
try {
    new StorageInitializer().createODKDirs();
} catch (RuntimeException e) {
    createErrorDialog(e.getMessage(), EXIT);
    return;
}
```

For this to work the app's "storage state" has to be setup correctly. It wasn't so the `catch` branch was firing before `adminSharedPreferences` was initialized. I think it would nice to rework this whole test (which is something I can take a look) so for the moment could you create a new `MainMenuActivityTest` with your test (iwith a new name and without the javadoc comment). The `@Before` you need to make sure storage state is setup is:

```java
@Before
public void setup() {
    ShadowEnvironment.setExternalStorageState(MEDIA_MOUNTED);
    mainMenuActivity = Robolectric.setupActivity(MainMenuActivity.class);
}
```